api/RS.API/RS.Data/Logic/SkillRepository.cs

```diff
@@ -2,19 +2,57 @@
 using RS.Entity.Models;
 using System.Linq;
 using System.Collections.Generic;
+using RS.ViewModel.Skill;
+using System;
+using RS.ViewModel.SearchAndSortModel;
+using RS.Common.Extensions;
 
 namespace RS.Data.Logic
 {
     public class SkillRepository : Repository<Skills>, ISkillRepository
     {
         private readonly RSContext _context;
+
         public SkillRepository(RSContext context) : base(context) {
 
             this._context = context;
         }
-        List<Skills> ISkillRepository.GetAll()
+
+        List<SkillViewModel> ISkillRepository.GetAll(SearchAndSortModel searchAndSortModel)
         {
-            return _context.Skills.Where(x => (x.IsActive && !x.IsDeleted)).ToList();
+
+            List<SkillViewModel> skillModelList = new List<SkillViewModel>();
+            var GetAll = _context.Skills.Where(x => (x.IsActive && !x.IsDeleted)).ToList();
+           
+            var skillLists = skillModelList.MapFromModel<Skills, SkillViewModel>(GetAll);
+            List<SkillViewModel> skillOrderList = new List<SkillViewModel>();
+            List<SkillViewModel> skillSearchSort = new List<SkillViewModel>();
```

Comment:
No need to use these variables. Use skillLists for search and sort operation results